JASP-Engine/JASP/R/binomialtest.R

```diff
@@ -15,224 +15,315 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-BinomialTest <- function(dataset, options, perform, state, ...) {
+BinomialTest <- function(jaspResults, dataset, options, ...) {
 
-	variables <- unlist(options$variables)
+  # Set title
+  jaspResults$title <- "Binomial Test"
 
-	binomResults <- state$binomResults
-  if (is.null(binomResults))
-		binomResults <- .binomialTest(dataset, options, perform, variables)
+  # Read dataset
+  dataset <- .binomReadData(dataset, options)
 
-	descriptPlots <- state$descriptPlots
-  if (options[["descriptivesPlots"]] && is.null(descriptPlots))
-		descriptPlots <- .binomialDescriptivesPlot(dataset, options, perform, variables)
+  # Error checking
+  .binomCheckErrors(dataset, options)
 
-	results <- list()
-	results[["binomial"]] <- .BinomialTable(binomResults, options)
+  # Compute the results
+  binomResults <- .binomComputeResults(jaspResults, dataset, options)
 
-	if (! is.null(descriptPlots)) {
-		results[["descriptives"]] <- descriptPlots
-		results[["descriptives"]][["title"]] <- "Descriptive Plots"
-	}
+  # Output tables and plots
+  .binomTableMain(jaspResults, dataset, options, binomResults)
+  .binomContainerPlots(jaspResults, dataset, options, binomResults)
+  .binomPlotsDescriptive(jaspResults, dataset, options, binomResults)
 
-  if (perform == "init")
-		return(list(results=results, status="inited", state=state))
+  return()
 
-	state <- list(
-		options = options,
-		binomResults = binomResults,
-		descriptPlots = descriptPlots
-	)
+}
 
-  return(list(results=results, status="complete", state=state))
+# Preprocessing functions ----
+.binomReadData <- function(dataset, options) {
+  if (!is.null(dataset)) {
+    return(dataset)
+  } else {
+    return(.readDataSetToEnd(columns.as.factor = options$variables))
```

Comment:
I'm a bit confused. You read the data in as factor, but have a lot of `as.factor` down the road. Does this mean that the reading of the data with `columns.as.factor` doesn't guarantee the data are factors? If yes, the `as.factor` calls down the road are superfluous.

Reply:
this is probably true and an artefact of developing in RStudio using the original dataset without it having gone through the data loading pipeline of JASP. I'll look into it.

Reply:
I have removed as.factor inside the functions